File: implementations/react-native/design-patterns-rn/src/patterns/behavioral/memento/memento.ts
export class EditorMemento {
  readonly text: string;
  readonly createdAt: Date;

  constructor(text: string, createdAt = new Date()) {
    this.text = text;
    this.createdAt = createdAt;
  }

  getState() {
    return { text: this.text, createdAt: this.createdAt };
  }
}

File: implementations/flutter/design_patterns_flutter/lib/patterns/behavioral/memento/memento.dart
class EditorMemento {
  final String text;
  final DateTime createdAt;
  EditorMemento(this.text, [DateTime? created])
    : createdAt = created ?? DateTime.now();

  Map<String, Object> getState() => {'text': text, 'createdAt': createdAt};
}
